target/scripts/startup/setup.d/mail_state.sh

```diff
@@ -83,6 +83,10 @@ function _setup_save_states
     chown root:root /var/mail-state/spool-postfix
     # These two require the postdrop(103) group:
     chgrp -R postdrop /var/mail-state/spool-postfix/{maildrop,public}
+    # After changing the group, special bits (set-gid, sticky) may be stripped, restore them:
+    # Ref: https://github.com/docker-mailserver/docker-mailserver/pull/3149#issuecomment-1454981309
+    chmod 1730 /var/spool/postfix/maildrop
+    chmod 2710 /var/spool/postfix/public
```

Comment:
Should probably match `/var/mail-state/spool-postfix/` path directly like the other lines do?

```suggestion
    chmod 1730 /var/mail-state/spool-postfix/maildrop
    chmod 2710 /var/mail-state/spool-postfix/public
```